backend/battles/views.py

```diff
@@ -39,10 +43,28 @@ def get_context_data(self, **kwargs):
         user = User.objects.get(id=self.request.user.id)
         battle = self.kwargs["pk"]
         if user.teams.filter(battle=battle).exists():
-            context["user_has_team"] = True
-
+            messages.info(
+                self.request,
+                "You've chosen your team for this battle. Check your email for results",
+            )
         return context
 
+    def form_valid(self, form):
+        self.object = form.save()
+        battle = self.object.battle
+        creator = battle.user_creator
+        opponent = battle.user_opponent
+
+        if self.object.trainer == battle.user_creator:
+            send_invite_to_match(
```

Comment:
I would recommend you sending this invitation to an async task.

Reply:
Makes sense, will make those changes